### PR TITLE
(#113) Replace remaining uses Assertion -> Assertion2

### PR DESCRIPTION
**Pull request by llorllale**
PR for #113:
* Replaced all remaining uses of `Assertion` with `Assertion2`
* Refactored a few tests to use `Matches`
* Added a couple of missing tests
* Left puzzle to rename `Assertion2` back to `Assertion` (overwrite) and finish this transition

### CODE REVIEW

**Comment by 0crat** [comment]
Job [#115](https://github.com/llorllale/cactoos-matchers/issues/115) is now in scope, role is `REV`

**Comment by 0crat** [comment]
This pull request [#115](https://github.com/llorllale/cactoos-matchers/issues/115) is assigned to @ilyakharlamov[/z](https://www.0crat.com/u/ilyakharlamov), here is [why](https://www.0crat.com/footprint/C63314D6Z/4ecc1e41-1f0c-4f16-94a8-7abca04de6a4); the budget is 15 minutes, see [§4](http://www.zerocracy.com/policy.html#4); please, read [§27](http://www.zerocracy.com/policy.html#27) and when you decide to accept the changes, inform @llorllale[/z](https://www.0crat.com/u/llorllale) (the architect) right in this ticket; if you decide that this PR should not be accepted ever, also inform the architect; this [blog post](http://www.yegor256.com/2015/02/09/serious-code-reviewer.html) will help you understand what is expected from a code reviewer; there will be no monetary reward for this job

File: src/test/java/org/llorllale/cactoos/matchers/MatchesTest.java [hunk 1]
```diff
@@ -48,9 +48,9 @@
      */
     @Test
     public void matches() {
-        new Assertion<>(
+        new Assertion2<>(
```
Comment (ilyakharlamov) [inline src/test/java/org/llorllale/cactoos/matchers/MatchesTest.java]: `@checkstyle MagicNumberCheck` can be removed I believe

File: src/test/java/org/llorllale/cactoos/matchers/HasEntryTest.java [hunk 2]
```diff
@@ -75,9 +75,9 @@ public void doesNotMatchAMissingEntry() {
 
     @Test(expected = AssertionError.class)
```
Comment (ilyakharlamov) [inline src/test/java/org/llorllale/cactoos/matchers/HasEntryTest.java]: need to get rid of Junit4-style assertions and use `Throws`

File: src/test/java/org/llorllale/cactoos/matchers/HasEntryTest.java [hunk 3]
```diff
@@ -63,9 +63,9 @@ public void matches() {
...
```
Comment (ilyakharlamov) [inline src/test/java/org/llorllale/cactoos/matchers/HasEntryTest.java]: get rid of Junit4-style assertions and use Throws

File: src/test/java/org/llorllale/cactoos/matchers/HasEntryTest.java (hunk 2 above)
Comment (llorllale) [reply src/test/java/org/llorllale/cactoos/matchers/HasEntryTest.java]: @ilyakharlamov doing that reduces the coverage which means this PR will be blocked. There is a puzzle open for this at the top of this file. See https://github.com/llorllale/cactoos-matchers/pull/105#issuecomment-486203698

File: src/test/java/org/llorllale/cactoos/matchers/HasEntryTest.java (hunk 3 above)
Comment (llorllale) [reply src/test/java/org/llorllale/cactoos/matchers/HasEntryTest.java]: @ilyakharlamov doing that reduces the coverage which means this PR will be blocked. There is a puzzle open for this at the top of this file. See https://github.com/llorllale/cactoos-matchers/pull/105#issuecomment-486203698

File: src/test/java/org/llorllale/cactoos/matchers/MatchesTest.java (hunk 1 above)
Comment (llorllale) [reply src/test/java/org/llorllale/cactoos/matchers/MatchesTest.java]: @ilyakharlamov ok

**Comment by llorllale** [comment]
@ilyakharlamov please check

**Comment by ilyakharlamov** [comment]
@llorllale looks good to me

**Comment by llorllale** [comment]
@rultor merge

**Comment by 0crat** [comment]
Job was finished in 24 hours, bonus for fast delivery is possible (see [§36](http://www.zerocracy.com/policy.html#36))

**Comment by 0crat** [comment]
Job [#115](https://github.com/llorllale/cactoos-matchers/issues/115) is not in the agenda of @ilyakharlamov[/z](https://www.0crat.com/u/ilyakharlamov), can't inspect

**Comment by 0crat** [comment]
@sereshqua[/z](https://www.0crat.com/u/sereshqua) please review this job completed by @ilyakharlamov[/z](https://www.0crat.com/u/ilyakharlamov), as in [§30](http://www.zerocracy.com/policy.html#30); the job will be fully closed and all payments will be made when the quality review is completed

**Comment by 0crat** [comment]
The job [#115](https://github.com/llorllale/cactoos-matchers/issues/115) is now out of scope

**Comment by 0crat** [comment]
Payment to `ARC` for a closed pull request, as in [§28](http://www.zerocracy.com/policy.html#28): +10 point(s) just awarded to @llorllale[/z](https://www.0crat.com/u/llorllale)

**Comment by sereshqua** [comment]
@0crat quality good

**Comment by 0crat** [comment]
Quality review completed: +4 point(s) just awarded to @sereshqua[/z](https://www.0crat.com/u/sereshqua)

**Comment by 0crat** [comment]
Order was finished, quality is "good": +25 point(s) just awarded to @ilyakharlamov[/z](https://www.0crat.com/u/ilyakharlamov)

**Pull request merged**